src/compiler/checker.ts

```diff
@@ -28040,6 +28036,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return getUnionType(map(clauseWitnesses, text => text ? narrowTypeByTypeName(type, text) : neverType));
         }
 
+        function narrowTypeBySwitchOnTrue(type: Type, switchStatement: SwitchStatement, clauseStart: number, clauseEnd: number): Type {
+            // Equal start and end denotes implicit fallthrough; undefined marks explicit default clause.
+            const defaultIndex = findIndex(switchStatement.caseBlock.clauses, clause => clause.kind === SyntaxKind.DefaultClause);

```

Comment:
```suggestion
            const defaultIndex = findIndex(switchStatement.caseBlock.clauses, isDefaultClause);

```

Reply:
This is just the code from the `switch (typeof x)` function above. I can change it but it'll be different.